src/isar/robot/robot.py

```diff
@@ -0,0 +1,90 @@
+
+
+class Robot(object):
+
+    @inject
+    def __init__(self, queues: Queues, robot: RobotInterface):
+        self.logger = logging.getLogger("robot")
+        self.queues: Queues = queues
+        self.robot: RobotInterface = robot
+        self.start_mission_thread: Optional[RobotStartMissionOrTaskThread] = None
+        self.robot_status_thread: Optional[RobotStatusThread] = None
+        self.robot_task_status_thread: Optional[RobotTaskStatusThread] = None
+        self.signal_thread_quitting: Event = Event()
+
+    def stop(self) -> None:
+        self.signal_thread_quitting.set()
+        if self.robot_status_thread is not None and self.robot_status_thread.is_alive():
+            self.robot_status_thread.join()
+        if (
+            self.robot_task_status_thread is not None
+            and self.robot_status_thread.is_alive()
+        ):
+            self.robot_task_status_thread.join()
+        if (
+            self.start_mission_thread is not None
+            and self.robot_status_thread.is_alive()
+        ):
+            self.start_mission_thread.join()
+        self.robot_status_thread = None
+        self.robot_task_status_thread = None
+        self.start_mission_thread = None
+        self.signal_thread_quitting.clear()
+
+    def handle_stop_mission(self):
+        return NotImplementedError()
+
+    def handle_pause_mission(self):
+        return NotImplementedError()
+
+    def run(self) -> None:
+        self.robot_status_thread = RobotStatusThread(
+            self.queues, self.robot, self.signal_thread_quitting
+        )
+        self.robot_status_thread.start()
+
+        while True:
+            if self.signal_thread_quitting.is_set():
+                break
+            start_mission_or_task = check_for_event(
+                self.queues.state_machine_start_mission
+            )
+            if start_mission_or_task is not None:
+                if (
+                    self.start_mission_thread is not None
+                    and self.start_mission_thread.is_alive()
+                ):
+                    self.start_mission_thread.join()
+                self.start_mission_thread = RobotStartMissionOrTaskThread(
+                    self.queues,
+                    self.robot,
+                    self.signal_thread_quitting,
+                    start_mission_or_task,
+                )
+                self.start_mission_thread.start()
+            if check_for_event(self.queues.state_machine_stop_mission):
+                if (
+                    self.start_mission_thread is not None
+                    and self.start_mission_thread.is_alive()
+                ):
+                    self.start_mission_thread.join()
+                self.handle_stop_mission()
+            if check_for_event(self.queues.state_machine_pause_mission):
+                self.handle_pause_mission()
+            task_id = check_for_event(self.queues.state_machine_get_task_status)
```

Comment:
Change state_machine_get_task_status to state_machine_request_new_task_status_event